src/VisualStudio/IntegrationTest/TestUtilities/InProcess/FindReferencesWindow_InProc.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using EnvDTE80;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Common;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess
+{
+    internal class FindReferencesWindow_InProc : InProcComponent
+    {
+        public static FindReferencesWindow_InProc Create() => new FindReferencesWindow_InProc();
+
+        public Reference[] GetContents(string windowCaption)
+        {
+            return InvokeOnUIThread(() =>
```

Comment:
I'm going to guess the answer is yes, but does all of this actually require executing on the UI thread? Also, does all of this need to execute in proc? The DTE should be accessible, for the most part, out of proc (in the test process).